yarn-project/p2p/src/mem_pools/tx_pool/aztec_kv_tx_pool.ts

```diff
@@ -300,4 +342,181 @@ export class AztecKVTxPool implements TxPool {
       }
     });
   }
+
+  /**
+   * Evicts pending txs with the lowest priority fees from the pool to accomodate the max tx count and cumulative max tx size
+   * after new txs are added.
+   *
+   * @param newTxHashes - The tx hashes of the new txs added to the pool.
+   * @returns The total number of txs evicted from the pool and the number of new txs that were evicted.
+   *
+   * TODO: add logging for evicted txs?
+   */
+  private async evictLowPriorityTxs(
+    newTxHashes: TxHash[],
+  ): Promise<{ numLowPriorityTxsEvicted: number; numNewTxsEvicted: number }> {
+    let pendingTxsByPriority = (await toArray(this.#pendingTxPriorityToHash.valuesAsync())).map(TxHash.fromString);
```

Comment:
The `toArray` will mean we go through all tx hashes every time this function is called, which could be very often since it's being called from `addTxs`. Let's use the iterator to go lazily over the list.